src/node_i18n.cc

```diff
@@ -185,13 +186,92 @@ static void ToASCII(const FunctionCallbackInfo<Value>& args) {
                           len).ToLocalChecked());
 }
 
+// This is similar to wcwidth except that it takes the current unicode
+// character properties database into consideration, allowing it to
+// correctly calculate the column widths of things like emoji's and
+// newer wide characters. wcwidth, on the other hand, uses a fixed
+// algorithm that does not take things like emoji into proper
+// consideration.
+static int GetColumnWidth(UChar32 codepoint, bool ambiguousFull = false) {
```

Comment:
Style: s/ambiguousFull/ambiguous_full/ here and elsewhere.  I don't like the name very much, it doesn't really convey what it does.